packages/astro/src/core/middleware/index.ts

```diff
@@ -85,10 +87,14 @@ function createContext({
 		},
 		url,
 		get clientAddress() {
-			if (clientAddressSymbol in request) {
-				return Reflect.get(request, clientAddressSymbol) as string;
+			if (clientIpAddress) {
+				return clientIpAddress;
 			}
-			throw new AstroError(AstroErrorData.StaticClientAddressNotAvailable);
+			clientIpAddress = getClientIpAddress(request);
```

Comment:
I assume this is done here so that we don't need to update each adapter to provide it?

Reply:
Exactly!